sync/src/main/kotlin/gropius/sync/AbstractSync.kt

```diff
@@ -510,16 +596,53 @@ abstract class AbstractSync(
     private suspend fun syncOutgoingLabels(
         timeline: List<TimelineItem>, imsProject: IMSProject, issueInfo: IssueConversionInformation
     ) {
-        val groups = timeline.filter { (it is AddedLabelEvent) || (it is RemovedLabelEvent) }.groupBy {
-            when (it) {
-                is AddedLabelEvent -> it.addedLabel().value
-                is RemovedLabelEvent -> it.removedLabel().value
-                else -> throw IllegalStateException()
+        val labelStateMap = this.labelStateMap(imsProject)
+        val stateLabelMap = labelStateMap.map { it.value to it.key }.toMap()
+        val virtualIDs = mutableMapOf<TimelineItem, String>()
+
+        val groups =
```

Comment:
maybe some temp variables would help splitting this into more readable chunks
that outer () spans a lot of lines

Reply:
1e4486afdb83b37ceade378275bf257fc60fdc35